src/main/java/com/thealgorithms/others/SieveOfEratosthenes.java

```diff
@@ -1,82 +1,48 @@
-package com.thealgorithms.others;
-
+import java.util.ArrayList;
 import java.util.Arrays;
 
-/**
- * Sieve of Eratosthenes is an ancient algorithm for finding all prime numbers
- * up to any given limit. It does so by iteratively marking as composite (i.e.,
- * not prime) the multiples of each prime, starting with the first prime number,
- * 2. The multiples of a given prime are generated as a sequence of numbers
- * starting from that prime, with constant difference between them that is equal
- * to that prime. This is the sieve's key distinction from using trial division
- * to sequentially test each candidate number for divisibility by each prime.
- * Once all the multiples of each discovered prime have been marked as
- * composites, the remaining unmarked numbers are primes.
- * <p>
- * Poetry about Sieve of Eratosthenes:
- * <p>
- * <i>Sift the Two's and Sift the Three's:</i></p>
- * <p>
- * <i>The Sieve of Eratosthenes.</i></p>
- * <p>
- * <i>When the multiples sublime,</i></p>
- * <p>
- * <i>The numbers that remain are Prime.</i></p>
- *
- * @see <a href="https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes">Wiki</a>
- */
 public class SieveOfEratosthenes {
 
     /**
      * @param n The number till which we have to check for prime Prints all the
      * prime numbers till n. Should be more than 1.
-     * @return array of all prime numbers between 0 to n
+     * @return ArrayList of all prime numbers between 0 to n
```

Comment:
Why do you return ArrayList instead of array?

Reply:
Arrays in Java have a fixed size, so if we were to return an array from the method, we would need to specify its size before calling the method. This can be inconvenient in some cases. ArrayLists, on the other hand, can grow dynamically as needed, so they are more flexible.

Reply:
Why do we need to specify the array size before calling the method? We can figure out how many elements in array we need inside the method. The biggest difference between them is that ArrayList can be mutated easier than array and we don't want that because it's only used to return data. So using array is better in this case. You can use ArrayList inside the method when accummulating primes, but return an array.